src/app/core/device.ts

```diff
@@ -0,0 +1,1044 @@
+     * @memberof Device
+     * @returns {void}
+     */
+    __loadLayersFromJSON(json: {[index: string]: any}): void {
+        for (let i in json) {
+            let newLayer = Layer.fromJSON(json[i]);
+            this.addLayer(newLayer);
+        }
+    }
+    /**
+     * Loads feature layers from a Interchange format into the device object
+     * @param {*} json 
+     * @memberof Device
+     * @returns {void}
+     */
+    __loadFeatureLayersFromInterchangeV1(json: Array<LayerInterchangeV1>): void {
+        for (let i in json) {
+            let newLayer = Layer.fromInterchangeV1(json[i]);
+            this.addLayer(newLayer);
+        }
+    }
+
+    /**
+     * Loads the JSON Component object into the device object
+     * @param {ComponentInterchangeV1} components
+     * @memberof Device
+     * @returns {void}
+     * @private
+     */
+    __loadComponentsFromInterchangeV1(components: Array<ComponentInterchangeV1>): void {
+        let componenttoadd;
+
+        for (let i in components) {
+            componenttoadd = Component.fromInterchangeV1(components[i]);
+            this.__components.push(componenttoadd);
+        }
+    }
+    /**
+     * Loads connections to the device object
+     * @param {Connection} connections Connections to add to the device
+     * @memberof Device
+     * @returns {void}
+     */
+    __loadConnectionsFromInterchangeV1(connections: Array<ConnectionInterchangeV1>): void {
+        let connectiontoload;
+        for (let i in connections) {
+            connectiontoload = Connection.fromInterchangeV1(this, connections[i]);
+            this.__connections.push(connectiontoload);
+        }
+    }
+    /**
+     * Converts the properties of the device to JSON 
+     * @returns {JSON} Returns a JSON format with the properties of the device
+     * @memberof Device
+     */
+    toJSON(): {[index: string]: any} {
+        let output: {[index: string]: any} = {};
+        output.name = this.__name.toJSON();
+        output.params = this.__params.toJSON();
+        output.layers = this.__layersToJSON();
+        output.groups = this.__groupsToJSON();
+        output.defaults = this.__defaults;
+        return output;
+    }
+    /**
+     * Converts to Interchange V1 format
+     * @returns {Device} Returns an Device object in Interchange V1 format
+     * @memberof Device
+     */
+    toInterchangeV1(): DeviceInterchangeV1 {
+        let output: DeviceInterchangeV1 = {
+            name = this.__name,
+            params = {
+                width: this.getXSpan(),
+                length: this.getYSpan()
+            },
+        //TODO: Use this to dynamically create enough layers to scroll through
+        // output.layers = this.__layersToInterchangeV1();
+            components = this.__componentsToInterchangeV1(),
+            connections = this.__connectionToInterchangeV1(),
+        //TODO: Use this to render the device features
+            features = this.__featureLayersToInterchangeV1(),
```

Comment:
'features' here takes on a LayerInterchangeV1 object. 'layers' remains empty. There is no property in DeviceInterchangeV1 specifically for Feature, unsure whether there should be.

Reply:
Yeah this needs to be rewritten a little

Reply:
Should I rename `features` `layers` here?